src/NuGet.Server.V2/Controllers/NuGetODataController.cs

```diff
@@ -50,24 +52,33 @@ protected NuGetODataController(IServerPackageRepository repository, IPackageAuth
         // Probably required to be OData compliant?
         [HttpGet]
         [EnableQuery(PageSize = 100, HandleNullPropagation = HandleNullPropagationOption.False)]
-        public virtual IHttpActionResult Get(ODataQueryOptions<ODataPackage> options)
+        public virtual async Task<IHttpActionResult> Get(
+            ODataQueryOptions<ODataPackage> options,
+            CancellationToken token)
         {
-            var sourceQuery = _serverRepository.GetPackages();
+            var sourceQuery = await _serverRepository.GetPackagesAsync(token);
+
             return TransformToQueryResult(options, sourceQuery);
         }
 
         // GET /Packages/$count
         [HttpGet]
-        public virtual IHttpActionResult GetCount(ODataQueryOptions<ODataPackage> options)
+        public virtual async Task<IHttpActionResult> GetCount(
+            ODataQueryOptions<ODataPackage> options,
+            CancellationToken token)
         {
-            return (Get(options)).FormattedAsCountResult<ODataPackage>();
+            return (await Get(options, token)).FormattedAsCountResult<ODataPackage>();
```

Comment:
I considered this, but I wanted to leave the controller actions with their existing names. There is mapping logic around the `$count` methods that would make changing these action names a little nasty. So I thought the cost outweighed the benefit.